experiment/config-forker/main.go

```diff
@@ -313,63 +313,237 @@ func generateNameVariant(name, version string, generic bool) string {
+			}
+		}
+
+		if masterStart != -1 && masterStop == -1 && j > masterStart {
+			if countLeadingSpace(string(line)) <= spacesForBranch {
+				masterStop = j - 1
+			}
+		}
 	}
 
-	newPresubmits, err := generatePresubmits(c, o.newVersion)
-	if err != nil {
-		log.Fatalf("Failed to generate presubmits: %v.\n", err)
+	if masterStart != -1 && masterStop == -1 {
+		masterStop = len(branchLines) - 1
 	}
-	newPeriodics, err := generatePeriodics(c, o.newVersion)
-	if err != nil {
-		log.Fatalf("Failed to generate periodics: %v.\n", err)
+
+	if masterStart != -1 && masterStop != -1 {
+
+		contextStart := -1
+		contextStop := -1
+		contextStrings := -1
+
+		secondBranchSpaces := countLeadingSpace(branchLines[masterStart])
+
+		for m := 0; m < masterStop-masterStart+1; m++ {
+			masterLine := branchLines[m+masterStart]
+			if strings.Contains(masterLine, "contexts:") {
+				contextStart = m + 1
+				contextStrings = countLeadingSpace(masterLine)
+			}
+
+			if contextStart != -1 && contextStrings != -1 && contextStop == -1 && m > contextStart {
+
+				if strings.Compare(string(masterLine[contextStrings]), "-") != 0 {
+					contextStop = m
+				}
+			}
+		}
+
+		if contextStart != -1 && contextStop == -1 {
+			contextStop = masterStop - masterStart + 1
+		}
+		containsNeedsAdmin := false
+		newBranchSlice := copySlice(branchLines[masterStart : masterStop+1])
+		if contextStart != -1 && contextStop != -1 {
+			for n := contextStart; n < contextStop; n++ {
+				contextLine := branchLines[n+masterStart]
+
+				if strings.Contains(contextLine, "merges-blocked-needs-admin") {
+					containsNeedsAdmin = true
+					break
+				}
+			}
+
+			if !containsNeedsAdmin {
+				numSpaces := countLeadingSpace(newBranchSlice[contextStart])
+				stringAdmin := strings.Repeat(" ", numSpaces) + "- \"merges-blocked-needs-admin\""
+
+				newBranchSlice = appendAtIndex(newBranchSlice, stringAdmin, contextStop-1)
+			}
+
+		} else {
+
+			if len(string(newBranchSlice[len(newBranchSlice)-1])) == countLeadingSpace(string(newBranchSlice[len(newBranchSlice)-1])) {
+				newBranchSlice = newBranchSlice[:len(newBranchSlice)-1]
+			}
+			statusCheckString := strings.Repeat(" ", secondBranchSpaces) + "required_status_checks:"
+			contextsString := strings.Repeat(" ", secondBranchSpaces+2) + "contexts:"
+			needsAdminString := strings.Repeat(" ", secondBranchSpaces+2) + "- \"merges-blocked-needs-admin\""
+			newBranchSlice = append(newBranchSlice, statusCheckString, contextsString, needsAdminString)
+		}
+
+		newBranchLine := strings.Repeat(" ", spacesForBranch) + newBranch + ":"
+		newBranchSlice = appendAtIndex(newBranchSlice, newBranchLine, 0)
+		masterBranch := strings.Join(newBranchSlice, "\n")
+		return masterStart - 1, masterBranch
 	}
-	newPostsubmits, err := generatePostsubmits(c, o.newVersion)
-	if err != nil {
-		log.Fatalf("Failed to generate postsubmits: %v.\n", err)
+	return -1, ""
+}
+
+// Convert read result into string and split it based on "branches:" to make it
```

Comment:
This doesn't seem like a good way to parse and process yaml files. Why do you want/need to manipulate strings rather than use a parser?

Reply:
Thank you for the comment. I had to parse the config as a string because when I use the yaml parser for unmarshal and marshal, it escapes the double quotation marks from the original file and the pointers in the original files get expended in the process which would result in more diffs than I would like to see. Therefore I'm parsing it as strings.

Reply:
Unfortunately I do not think string parsing is a design we are likely to accept upstream. Can you try and work with @Katharine to come up with a better solution?